hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/RequestFactoryImpl.java

```diff
@@ -517,12 +518,22 @@ public CreateMultipartUploadRequest.Builder newMultipartUploadRequestBuilder(
   public CompleteMultipartUploadRequest.Builder newCompleteMultipartUploadRequestBuilder(
       String destKey,
       String uploadId,
-      List<CompletedPart> partETags) {
+      List<CompletedPart> partETags,
+      PutObjectOptions putOptions) {
+
     // a copy of the list is required, so that the AWS SDK doesn't
     // attempt to sort an unmodifiable list.
-    CompleteMultipartUploadRequest.Builder requestBuilder =
-        CompleteMultipartUploadRequest.builder().bucket(bucket).key(destKey).uploadId(uploadId)
+    CompleteMultipartUploadRequest.Builder requestBuilder;
+    Map<String, String> optionHeaders = putOptions.getHeaders();
```

Comment:
Add a boolean flag to put options, so that it can be enabled in different bits of the code as appropriate...matters for future features

Reply:
Could you please provide some additional details? I took a stab at it [here](https://github.com/diljotgrewal/hadoop/commit/0cf89eb5a06ec5b8be83af41aee07d8c2bed959e) but I'm not sure if I'm on the right track. 

Thanks!!

Reply:
added bool `conditionalPutEnabled` to PutObjectOptions